sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
+
+    /**
+     * Returns the total amount of parallelism in the consumed (and processed) range of this
+     * reader's current {@link BoundedSource} (as would be returned by {@link #getCurrentSource}).
+     *
+     * <p>The "parallelism" consumed does not necessarily equal the number of <em>records</em>
+     * output and processed, but rather the number of distinct (ideally non-empty) <em>sources</em>
+     * that could be produced by calls to {@link #splitAtFraction}.
+     *
+     * <p>Consider the following two examples. An input that can be read in parallel down to the
+     * individual records, such as {@link CountingSource#upTo}, is called "perfectly splittable".
+     * Conversely, an example of a non-perfectly splittable input is a block-compressed file format
+     * such as {@link AvroIO}, in which a block of records has to be read as a whole, but different
+     * blocks can be read in parallel.
+     *
+     * <ul>
+     * <li>Any {@link BoundedReader reader} that is unstarted (aka, has never had a call to
+     * {@link #start}) has a consumed parallelism of 0.
+     * <li>Any {@link BoundedReader reader} that is currently processing its first element (aka,
+     * has never had a call to {@link #advance}) has a consumed parallelism of 0: the first element
+     * is still being processed, and thus has not been consumed.
+     * <li>When processing record #30 (starting at 1) out of 50 in a perfectly splittable 50-record
+     * input, this value should be 29.
+     * <li>If we are reading through block 3 in a block-compressed file consisting of 5 blocks,
+     * this value should be 2.
+     * <li>For an empty reader (in which the call to {@link #start} returned false), the
+     * consumed parallelism is 0.
+     * <li>For a non-empty, finished reader (in which the call to {@link #start} returned true and
+     * a call to {@link #advance} has returned false), the value returned is at must be at least 1
+     * and should equal the total parallelism in the source. For the two examples above, the
+     * 50-record file should have a consumed parallelism of 50.
+     * </ul>
+     *
+     * <p>A reader that is implemented using a {@link org.apache.beam.sdk.io.range.RangeTracker} is
+     * encouraged to delegate this function to its {@code getParallelismConsumed} method, if
+     * implemented.
+     *
```

Comment:
Updated the specific comment to be more accurate. Specifically, suggests using a range tracker's ability to count split points and then link to `OffsetBasedReader`. We can't actually force a change to the RangeTracker API (backwards incompatible), so instead it makes sense to give a concrete example.